src/deploy/table_contact_policy.sql

```diff
@@ -12,34 +13,52 @@ GRANT INSERT, UPDATE, DELETE ON TABLE maevsi.contact TO maevsi_account;
 
 ALTER TABLE maevsi.contact ENABLE ROW LEVEL SECURITY;
 
--- Only display contacts referencing the invoker's account.
--- Only display contacts authored by the invoker's account.
--- Only display contacts for which an accessible invitation exists.
+-- Only display contacts referencing the invoker's account, omit contacts authored by a blocked account.
+-- Only display contacts authored by the invoker's account, omit contacts referring to a blocked account.
+-- Only display contacts for which an accessible invitation exists, omit contacts auhored by a blocked account or referring to a blocked account.
 CREATE POLICY contact_select ON maevsi.contact FOR SELECT USING (
   (
-    NULLIF(current_setting('jwt.claims.account_id', true), '')::UUID IS NOT NULL
```

Comment:
I think removing this `IS NOT NULL` check allows anonymous users to see all contacts that have no account id assigned, no?

Reply:
No, this is not correct. If in comparison one of the operands is null the comparison is considered false.